### PR TITLE
Validation as newtype over disjunction for scalaz 8

### PR DESCRIPTION
**Pull request by Jacoby6000**
There may be things wrong with this, but I'm not sure... I don't see where tests are written for other things, but I'd be happy to write some for this if people could point me in the right direction.

To write this, I basically just followed the patterns that the `Maybe` newtype is using.  While doing that, I noticed that the unapply methods on `Maybe` (#1832) will not get proper exhaustivity warnings.  There may have been discussion over this already, but I'm not aware.  Just want to make sure people are aware.   Also, because I copied `Maybe`, the `Validation` implementation has the same issue.

### CODE REVIEW

File: base/shared/src/main/scala/scalaz/data/validation.scala [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    case Invalid(a) => ia(a)
+    case Valid(b)   => vb(b)
+  }
+
+  @inline def fromDisjunction[L, R](disj: L \/ R): Validation[L, R] = disj
+
+  @inline def invalid[L, R](value: L): Validation[L, R] = -\/(value)
+  @inline def valid[L, R](value: R): Validation[L, R]   = \/-(value)
+
+  implicit def validationApplicative[L](implicit L: Semigroup[L]): Applicative[Validation[L, ?]] =
```
Comment (hrhino) [inline base/shared/src/main/scala/scalaz/data/validation.scala]: I think these instances aren't going to get resolved where they are. You need a (`ValidationInstances` mixed into a) `ValidationModule` object, is what I think we're doing.

Comment (Jacoby6000) [reply base/shared/src/main/scala/scalaz/data/validation.scala]: Ah right. Forgot that part.  Will add it

Comment (Jacoby6000) [reply base/shared/src/main/scala/scalaz/data/validation.scala]: Looks like we're doing it a little differently for newts, but you were write about the discoverability problem

**Comment by edmundnoble** [comment]
Now that we have #1910, the instances should go in a different place @Jacoby6000, otherwise I'm :+1: for merge.

**Comment by Jacoby6000** [comment]
@edmundnoble I actually just rethought this. I did this wrong... Instead, `Validation` should just be a tag that goes on `Either`, which prompts the compiler to use a different `Applicative` instance.. Maybe add syntactic sugar for something which isn't called `flatMap`, but has the same signature... `cats` has `andThen` on validation, for example.

**Pull request closed without merging**